src/OpenLoco/src/GameCommands/Vehicles/VehicleRearrange.cpp

```diff
@@ -0,0 +1,181 @@
+#include "VehicleRearrange.h"
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "VehiclePickupAir.h"
+#include "VehiclePickupWater.h"
+#include "Vehicles/Vehicle.h"
+#include "Vehicles/VehicleManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    struct PlacementBackup
+    {
+        int16_t tileX;
+        int16_t tileY;
+        World::SmallZ tileBaseZ;
+        Vehicles::TrackAndDirection trackAndDirection;
+        uint16_t subPosition;
+        EntityId head;
+    };
+
+    // 0x004AF4D6
+    static void sub_4AF4D6(Vehicles::VehicleBogie& source, Vehicles::VehicleBogie& dest)
+    {
+        registers regs{};
+        regs.esi = X86Pointer(&source);
+        regs.edi = X86Pointer(&dest);
+        call(0x004AF4D6, regs);
+    }
+
+    static currency32_t vehicleRearrange(const VehicleRearrangeArgs& args, uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        auto* sourceVehicle = EntityManager::get<Vehicles::VehicleBase>(args.source);
+        auto* destVehicle = EntityManager::get<Vehicles::VehicleBase>(args.dest);
+
+        if (!(flags & Flags::apply))
+        {
+            if (!sub_431E6A(sourceVehicle->owner))
+            {
+                return FAILURE;
+            }
+            if (!sub_431E6A(destVehicle->owner))
+            {
+                return FAILURE;
+            }
+
+            auto* sourceHead = EntityManager::get<Vehicles::VehicleHead>(sourceVehicle->getHead());
+            auto* destHead = EntityManager::get<Vehicles::VehicleHead>(destVehicle->getHead());
+            if (!sourceHead->canBeModified())
+            {
+                return FAILURE;
+            }
+            if (!destHead->canBeModified())
+            {
+                return FAILURE;
+            }
+
+            if (sourceVehicle->getTrackType() != destVehicle->getTrackType())
+            {
+                setErrorText(StringIds::incompatible_vehicle);
+                return FAILURE;
+            }
+
+            // Pretty sure this needs to be true but not 100% (openloco addition)
+            auto* sourceBogie = sourceVehicle->asVehicleBogie();
+            if (sourceBogie == nullptr)
+            {
+                return FAILURE;
+            }
+
+            if (!destHead->isVehicleTypeCompatible(sourceBogie->objectId))
+            {
+                return FAILURE;
+            }
+
+            setPosition(destVehicle->position);
+            return 0;
+        }
+        else
+        {
+            auto* sourceBogie = sourceVehicle->asVehicleBogie();
+            if (sourceBogie == nullptr)
+            {
+                return FAILURE;
+            }
+            auto* destBogie = destVehicle->asVehicleBogie();
+            if (destBogie == nullptr)
```

Comment:
I had a thought about this last night. Destination could be just a head vehicle.  It's only source that will always be a bogie (car)